### ISSUE TITLE
Honor process.env.PORT variable just like process.env.NODE_ENV

### ISSUE THREAD

**Issue by kepikoi** [vuejs/vue-cli#2839]
### What problem does this feature solve?
PaaS Platforms like Heroku use a dynamically assigned PORT setting to run node.js applications. This variable is available as process.env.PORT inside the node instance.

The vm should be able to read this setting to know what port to use to access its API  just like it's able to read the NODE_ENV variable. Using something like VUE_APP_PORT is not a substitute since it's not possible to predetermine the port value in the PaaS context. 

### What does the proposed API look like?
start the nodejs application from the cli:

``PORT=1234  node app.js``

then inside any Vue component:

``mounted() {
 console.log(process.env.PORT);  //-> 1234
}``

**Comment by LinusBorg** [comment]
You have tow choices:

### A. export the environment variable with `VUE_APP_*`
```
export VUE_APP_PORT=8081
```
Of course that only works if you can control the name of the variable that e.g. Heroku exposes. If you can't, then you can do the second way:

### B. Reassign it in `vue.config.js`

```javascript
process.env.VUE_APP_PORT = process.env.PORT

module.exports = {
  lintonsave: false
}
```

If you go either of these roads, you can access the port in your app via `process.env.VUE_APP_PORT`.

And for local development you can either provide a port through a `.env.local` file, or make a fallback in `vue.config.js`:
```javascript
process.env.VUE_APP_PORT = process.env.PORT || 8080
```

### Bottom Line

There's enough flexibility in the APIs that exist, and I don't think we should implement any process that treats some system env variables different than others. Users can define the desired behaviour completely on their own as shown above.
